dotCMS/src/main/java/com/dotcms/storage/StoragePersistenceProvider.java

```diff
@@ -99,15 +110,22 @@ private void initializeStorageChain() {
      *                            property has not been set.
      */
     private void initializeStorageChain(final String chainName, final String[] defaultStorageTypes) {
-
         final String[] storageTypes = Config.getStringArrayProperty(chainName, defaultStorageTypes);
         final ChainableStoragePersistenceAPIBuilder builder = new ChainableStoragePersistenceAPIBuilder();
-
         Arrays.stream(storageTypes).iterator().forEachRemaining(storageTypeName -> {
-            final StorageType defaultStorageType = StorageType.valueOf(storageTypeName);
-            builder.add(this.getStorage(defaultStorageType));
+            final StorageType storageType = StorageType.valueOf(storageTypeName);
+            if (isValidLicense()) {
+                builder.add(this.getStorage(storageType));
+            } else {
+                if (storageType.equals(StorageType.FILE_SYSTEM) || storageType.equals(StorageType.DB)) {
```

Comment:
If there's a possibility that `storageType` could be null, I would suggest validating in different order, to avoid nullPointerExecptions.

`if (StorageType.FILE_SYSTEM.equals(storageType) || StorageType.DB.equals(storageType)) {`

Reply:
Done!